src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/SqlExtendedQueryTagStoreV2.cs

```diff
@@ -134,29 +134,63 @@ internal static AddExtendedQueryTagsInputTableTypeV1Row ToAddExtendedQueryTagsIn
 
         public override async Task DeleteExtendedQueryTagAsync(string tagPath, string vr, CancellationToken cancellationToken = default)
         {
-            using (SqlConnectionWrapper sqlConnectionWrapper = await ConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken))
-            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
-            {
-                V2.DeleteExtendedQueryTag.PopulateCommand(sqlCommandWrapper, tagPath, (byte)ExtendedQueryTagLimit.ExtendedQueryTagVRAndDataTypeMapping[vr]);
+            using SqlConnectionWrapper sqlConnectionWrapper = await ConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken);
+            using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand();
 
-                try
-                {
-                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
-                }
-                catch (SqlException ex)
+            sqlCommandWrapper.CommandText = @"
+SET NOCOUNT     ON
+SET XACT_ABORT  ON
+
+BEGIN TRANSACTION
```

Comment:
I like this is more clean. What will happen if the Tag was still being indexed by the Async job. Does it know that it needs to cancel after this foes through?

Also, do we know why the second transaction timed out, is there a overlocking issue we need to fix?

Reply:
I've reverted back to 2 transaction. The lock + status should prevent the re-indexing activity from attempting to re-index additional rows.
But yes, there may be an issue with too many locks. Tracking the bug here: https://microsofthealth.visualstudio.com/Health/_workitems/edit/87860